internal/pkg/manifest/workload.go

```diff
@@ -264,11 +264,38 @@ func (c *CommandOverride) ToStringSlice() ([]string, error) {
 	return out, nil
 }
 
+type advancedAliasSliceOrStringSliceOrString struct {
+	AdvancedAliases     []AdvancedAlias
+	StringSliceOrString stringSliceOrString
+}
```

Comment:
Is there a reason why you preferred this route vs something like this:
```go
type Alias struct {
   value string       // mutually exclusive with advanced
   advanced []MultiAlias
}

type MultiAlias struct {
    value string  // mutually exclusive with advanced
    advanced []AliasWithHostedZone
}

type AliasWithHostedZone struct {
   name string
   hostedZone string
}
```

Reply:
I think this way is slightly better because we only have one alias field in the manifest, and we have multiple fields that are `stringSliceOrString`. If we have multiple alias fields in the manifest and one of them uses `Alias` whereas one of them uses `AliasWithHostedZone` then I would agree with the example above.

Reply:
gotcha! 
OK sounds good, what do you think of just getting rid of `advancedAliasSliceOrStringSliceOrString` and instead have:

```go
type Alias struct {
	AdvancedAliases     []AdvancedAlias  // Mutually exclusive with Simple.
	Simple stringSliceOrString
}
```
My concern is that "advancedAliasSliceOrStringSliceOrString" is really difficult to read 😂 it took me a while to realize it's "advancedAliasSlice" OR "StringSliceOrString" vs. "advancedAliasSliceOrStringSlice" OR "String"